### DISCUSSION TITLE
Rust

### DISCUSSION THREAD

**Discussion by rkiyanchuk** [shopokodu/community#4] [Теми для подкасту]
Плануємо запросити пана Глюка, давши йому домашку написати LSP плагін для редактора [helix](https://github.com/helix-editor/helix).
Прогрес: https://github.com/helix-editor/helix/compare/master...andriigrynenko:helix:lsp-snippet-support

Книга по Rust [перекладається українською](https://github.com/rust-lang-ua/rustbook_ukrainian).

**Comment by malor** [comment]
# Особливості Rust

1. Сильна статична типізація.
2. Незмінність за замовченням.
3. Компілюється в машинний код. Швидкість на рівні C/C++.
4. Безпечна робота з пам'яттю без GC завдяки моделі володіння даними.
5. Алгебраїчні типи даних. Pattern matching.
6. Макроси.
7. Підтримка багатопоточного і асинхронного програмування.
8. Виклики в C/C++ і навпаки.
9. Вбудований пакетний менеджер.

# План обговорення

1. Представлення гостя.
3. Коротко про особливості Rust.
4. Модель володіння даними. Borrow checker.
5. Узагальнене програмування. Динамічний і статичний поліморфізм. Traits.
6. [Домашнє завдання](https://github.com/orgs/shopokodu/discussions/4) пана Глюка.
7. Пакетний менеджер і екосистема. Cargo / crates.io.
